diff -ru openssl-0.9.8l/Configure openssl-0.9.8l.my/Configure
--- openssl-0.9.8l/Configure	2009-11-05 13:07:06.000000000 +0100
+++ openssl-0.9.8l.my/Configure	2010-12-07 16:44:41.907666034 +0100
@@ -1059,7 +1059,7 @@
 
 my $IsMK1MF=scalar grep /^$target$/,@MK1MF_Builds;
 
-$IsMK1MF=1 if ($target eq "mingw" && $^O ne "cygwin" && !is_msys());
+#$IsMK1MF=1 if ($target eq "mingw" && $^O ne "cygwin" && !is_msys());
 
 $no_shared = 0 if ($fipsdso && !$IsMK1MF);
 
diff -ru openssl-0.9.8l/e_os2.h openssl-0.9.8l.my/e_os2.h
--- openssl-0.9.8l/e_os2.h	2005-12-18 19:57:07.000000000 +0100
+++ openssl-0.9.8l.my/e_os2.h	2010-12-07 16:46:09.887666003 +0100
@@ -264,7 +264,7 @@
 # define OPENSSL_IMPLEMENT_GLOBAL(type,name)			     \
 	extern type _hide_##name;				     \
 	type *_shadow_##name(void) { return &_hide_##name; }	     \
-	static type _hide_##name
+	type _hide_##name
 # define OPENSSL_DECLARE_GLOBAL(type,name) type *_shadow_##name(void)
 # define OPENSSL_GLOBAL_REF(name) (*(_shadow_##name()))
 #else
